crates/subspace-farmer-components/src/plotting.rs

```diff
@@ -184,3 +180,69 @@ where
         piece_indexes,
     })
 }
+
+async fn plot_pieces_in_batches_non_blocking<PR: PieceReceiver>(
+    in_memory_sector_scalars: &mut Vec<Scalar>,
+    sector_index: u64,
+    piece_receiver: &PR,
+    piece_indexes: &[PieceIndex],
+    cancelled: &AtomicBool,
+    piece_receiver_batch_size: usize,
+) -> Result<(), PlottingError> {
+    const MAX_PIECE_RECEIVER_BATCH_SIZE: usize = 60;
+
+    if piece_receiver_batch_size == 0 || piece_receiver_batch_size > MAX_PIECE_RECEIVER_BATCH_SIZE {
+        return Err(PlottingError::IncorrectPieceReceivingBatchSize);
+    }
```

Comment:
This bothers me a bit as well, it should really be checked during application startup rather than filing when pieces are requested, but we'll change it later anyway, so leaving as is.